### PR TITLE
MixedFeed v4

### PR DESCRIPTION
**Pull request by nlemoine**
Salut @ambroisemaupate ! 

Finally took some time to work on this. I've made quite some changes. It's not as big as it looks (change in coding standards, types, etc.)

Here's the PR summary:
- Use a PSR16 cache provider
- Update coding standards to PSR12
- Update dependencies
- Remove deprecated methods
- Remove unused/deprecated classes
- Add types
- PHP 8 support

Some more notes: 
- I was not able to test every provider because I don't have credentials for all of them but I guess you do if you built those :) I currently tested: instagram basic api, facebook page feed, twitter feed, twitter search.
- I didn't update the docs, sorry, I don't have much time right now

You'll find some comments directly in the code for additional notes on changes made.

### CODE REVIEW

File: src/Canonical/FeedItem.php [hunk 1]
```diff
@@ -1,281 +1,192 @@
...
+    protected string $message = '';
+
+    protected ?DateTime $dateTime = null;
+
+    /** @var string[] */
+    protected array $tags = [];
+
+    protected ?int $likeCount = null;
+
+    protected ?stdClass $raw = null;
```
Comment (nlemoine) [inline src/Canonical/FeedItem.php:34]: Canonical items are nice but sometimes, you might need the original response which contains some details that the standard FeeItem doesn't provide (In my MixedFeed use, I had to put back twitter links in tweets).

File: src/MixedFeed.php [hunk 2]
```diff
@@ -1,48 +1,41 @@
...
 use GuzzleHttp\Pool;
 use GuzzleHttp\Psr7\Response;
 use RZ\MixedFeed\Canonical\FeedItem;
 use RZ\MixedFeed\Exception\FeedProviderErrorException;
-use RZ\MixedFeed\MockObject\ErroredFeedItem;
 
 /**
  * Combine feed providers and sort them by date and time, descending or ascending.
  */
-class MixedFeed extends AbstractFeedProvider
```
Comment (nlemoine) [inline src/MixedFeed.php:14]: Extending AbstractFeedProvider didn't really make sense on the MixedFeed class. MixedFeed doesn't really behave like a provider.

File: src/MixedFeed.php [hunk 3]
```diff
@@ -51,64 +44,42 @@ public function __construct(array $providers = [], $sortDirection = MixedFeed::D
     }
 
     /**
-     * @deprecated
```
Comment (nlemoine) [inline src/MixedFeed.php:54]: Since it was deprecated in v3, I assumed the purpose was to remove it in v4.

**Comment by ambroisemaupate** [comment]
Thanks @nlemoine for your awesome work!

I'll check that as soon as I can.

**Comment by ambroisemaupate** [comment]
v4.0.0 is released thanks to your work @nlemoine.

**Comment by nlemoine** [comment]
Merci @ambroisemaupate !

**Pull request merged**